src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -2110,6 +2110,376 @@ void checkNodes()
             }
         }
 
+        [Fact, WorkItem(32106, "https://github.com/dotnet/roslyn/issues/32106")]
+        public void DuplicateAsyncs1()
+        {
+            const string text = """
+                class Program
+                {
+                    void M()
+                    {
+                        async async void F() { }
+                    }
+                }
+                """;
+
+            CreateCompilation(text).VerifyDiagnostics(
+                // (5,15): error CS1031: Type expected
+                //         async async void F() { }
+                Diagnostic(ErrorCode.ERR_TypeExpected, "async").WithLocation(5, 15),
+                // (5,15): error CS1004: Duplicate 'async' modifier
+                //         async async void F() { }
+                Diagnostic(ErrorCode.ERR_DuplicateModifier, "async").WithArguments("async").WithLocation(5, 15),
+                // (5,26): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
+                //         async async void F() { }
+                Diagnostic(ErrorCode.WRN_AsyncLacksAwaits, "F").WithLocation(5, 26),
+                // (5,26): warning CS8321: The local function 'F' is declared but never used
+                //         async async void F() { }
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "F").WithArguments("F").WithLocation(5, 26));
+
+            UsingDeclaration(text, options: TestOptions.Regular9,
+                // (5,15): error CS1031: Type expected
```

Comment:
>// (5,15): error CS1031: Type expected [](http://example.com/codeflow?start=16&length=38)

Why do we report this error? It looks like there is a type and it is in the syntax tree.

Reply:
For statements, when there is a duplicate modifier, the error "CS1031: Type expected" is reported during parsing. So code like `const const int a = 2;` reports "CS1031" on the second `const` modifier dispite the fact that the type is present and it is just modifier duplication ([sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCecAmIA1AD4ACATAIwCwAUGQMwAElzAwswN73N+ssyAFmYBZABQBKbr35yAxgHsYAZwTMlq9bHURmAXmYUA3LL4BfeuaA=)). Since local functions are statements, this error is reported for them as well. But then the "Duplicate '{0}' modifier" is reported during binding, so we get 2 errors on the same location.

There is no "fast" fix for this thing. When statement modifiers are parsed, the result statement kind is yet unknown. If we remove reporting of "CS1031: Type expected", we loose all errors for other statements. So we need to implement some kind of look ahead algorithm, or report duplicate modifiers error later after the statement is parsed or something else. This is not a trivial thing, so I left it here "as is". My intent in this PR is to produce correct tree for multiple `async`s. Then I'll make a follow up, where I'm gonna improve error reporting for these trees. So lets just left this thing for now to "freeze" current state.

Reply:
> Then I'll make a follow up, where I'm gonna improve error reporting for these trees. 

Let's open a follow up issue to track this work then

Reply:
Done, see https://github.com/dotnet/roslyn/issues/66077